### PR TITLE
feat(config): add default max marker as default values

### LINKED ISSUE #108: Implement proper config initialization for default values
For values that need a default other than the zero value (e.g., `20`), the configuration currently isn't initialized properly in tests, as the defaults would be set by cobra. Instead of adding the default values to each configuration that is being created in the tests, we should have a means to use a default configuration, e.g., by having a function `NewConfig`.

### PR DESCRIPTION
**Pull request by fzipi**
Signed-off-by: Felipe Zipitria <felipe.zipitria@owasp.org>

Fixes #108.

### CODE REVIEW

File: config/types.go [hunk 1]
```diff
@@ -17,14 +17,26 @@ const (
...
+	DefaultMaxMarkerLogLines int = 500
 )
 
 // FTWConfiguration FTW global Configuration
 type FTWConfiguration struct {
-	LogFile             string          `koanf:"logfile"`
-	TestOverride        FTWTestOverride `koanf:"testoverride"`
-	LogMarkerHeaderName string          `koanf:"logmarkerheadername"`
-	RunMode             RunMode         `koanf:"mode"`
+	// Logfile is the path to the file that contains the WAF logs to check
```
Comment (theseion) [inline config/types.go:28]: Can the path be relative? If so, relative to the working directory?

File: config/types.go [hunk 2]
```diff
@@ -17,14 +17,26 @@ const (
...
+	LogFile string `koanf:"logfile"`
+	// TestOverride holds the test overrides that will apply globally
+	TestOverride FTWTestOverride `koanf:"testoverride"`
+	// LogMarkerHeaderName is the name of the header that will be searched for on the logfile
```
Comment (theseion) [inline config/types.go:32]:
```suggestion
	// LogMarkerHeaderName is the name of the header that will be used by the test framework to mark positions in the log file
```

File: config/types.go [hunk 3]
```diff
@@ -17,14 +17,26 @@ const (
...
+	LogMarkerHeaderName string `koanf:"logmarkerheadername"`
+	// RunMode stores the mode used to deciding the results of the test. See https://github.com/coreruleset/go-ftw#%EF%B8%8F-cloud-mode.
```
Comment (theseion) [inline config/types.go:34]:
```suggestion
	// RunMode stores the mode used to interpret test results. See https://github.com/coreruleset/go-ftw#%EF%B8%8F-cloud-mode.
```

**Review by theseion — CHANGES_REQUESTED** [review]

File: config/types.go (hunk 1 above)
Comment (fzipi) [reply config/types.go:28]: Yes. Even in the examples listed in the README you have a relative path example.

**Review by theseion — APPROVED** [review]

**Pull request merged**